solidity/random-beacon/contracts/DKGValidator.sol

```diff
@@ -0,0 +1,230 @@
+    uint256 public constant groupThreshold = 33;
+
+    /// @dev The minimum number of active and properly behaving group members
+    ///      during the DKG needed to accept the result. This number is higher
+    ///      than `groupThreshold` to keep a safety margin for members becoming
+    ///      inactive after DKG so that the group can still produce a relay
+    ///      entry.
+    uint256 public constant activeThreshold = 58; // 90% of groupSize
+
+    SortitionPool public sortitionPool;
+
+    constructor(SortitionPool _sortitionPool) {
+        sortitionPool = _sortitionPool;
+    }
+
+    /// @notice Performs a full validation of DKG result, including checking the
+    ///         format of fields in the result, declared selected group members,
+    ///         and signatures of operators supporting the result.
+    /// @param seed seed used to start the DKG and select group members
+    /// @param startBlock DKG start block
+    /// @return isValid true if the result is valid, false otherwise
+    /// @return errorMsg validation error message; empty for a valid result
+    function validate(
+        DKG.Result calldata result,
+        uint256 seed,
+        uint256 startBlock
+    ) external view returns (bool isValid, string memory errorMsg) {
+        (bool hasValidFields, string memory error) = validateFields(result);
+        if (!hasValidFields) {
+            return (false, error);
+        }
+
+        if (!validateSignatures(result, startBlock)) {
+            return (false, "Invalid signatures");
+        }
+
+        if (!validateGroupMembers(result, seed)) {
+            return (false, "Invalid group members");
+        }
+
+        return (true, "");
+    }
+
+    /// @notice Performs a static validation of DKG result fields: lengths,
+    ///         ranges, and order of arrays.
+    /// @return isValid true if the result is valid, false otherwise
+    /// @return errorMsg validation error message; empty for a valid result
+    function validateFields(DKG.Result calldata result)
+        public
+        view
+        returns (bool isValid, string memory errorMsg)
+    {
+        // Group public key needs to be 128 bytes long.
+        if (result.groupPubKey.length != 128) {
+            return (false, "Malformed group public key");
+        }
+
+        // The number of misbehaved members can not exceed the threshold.
+        // Misbehaved member indices needs to be unique, between [1,64],
+        // and sorted in ascending order.
+        uint8[] calldata misbehavedMembersIndices = result
+            .misbehavedMembersIndices;
+        if (groupSize - misbehavedMembersIndices.length < activeThreshold) {
+            return (false, "Too many members misbehaving during DKG");
+        }
+        if (misbehavedMembersIndices.length > 1) {
+            if (
+                misbehavedMembersIndices[0] < 1 ||
+                misbehavedMembersIndices[misbehavedMembersIndices.length - 1] >
+                64
```

Comment:
Maybe `groupSize` instead of just `64`?

Reply:
https://github.com/keep-network/keep-core/pull/2734/commits/5bc95364647baabb9b6c7951805b9fb2681dcdfb